src/SurfaceFluxes.jl

```diff
@@ -616,18 +616,16 @@ function heat_exchange_coefficient(
     scheme,
     tol_neutral,
 )
-    FT = eltype(L_MO)
     thermo_params = SFP.thermodynamics_params(param_set)
     transport = UF.HeatTransport()
     κ = SFP.von_karman_const(param_set)
-    grav::FT = SFP.grav(param_set)
-    cp_d::FT = SFP.cp_d(param_set)
+    grav = SFP.grav(param_set)
     DSEᵥ_sfc = TD.virtual_dry_static_energy(thermo_params, ts_sfc(sc), grav * z_sfc(sc))
     DSEᵥ_in = TD.virtual_dry_static_energy(thermo_params, ts_in(sc), grav * z_in(sc))
     ΔDSEᵥ = DSEᵥ_in - DSEᵥ_sfc
     z0_b = z0(sc, UF.HeatTransport())
     z0_m = z0(sc, UF.MomentumTransport())
-    if isapprox(ΔDSEᵥ, FT(0); atol = tol_neutral)
```

Comment:
Not an objection, but I am wondering if this is equivalent. (I could see `isapprox` behaving differently between floats and Ints)

Reply:
Looking at the output of `code_llvm` and `code_native`, it seems like the overall code is the same, but using an`Int` adds a few promotion-related instructions. We can also rewrite this as `abs(ΔDSEᵥ) <= tol_neutral` to use even fewer instructions.

Reply:
I could see the gpu compiler liking `abs(ΔDSEᵥ) <= tol_neutral` more than `isapprox` tbh